### PR TITLE
ParamStore list_values(key) to get values of key across all commits

### PR DESCRIPTION
**Pull request by urig**
This PR implements the `list_values(key)` `ParamStore` method in `InProcessParamStore`.

Given a specific key, the method returns a list of its previous (and current) values across commits in the store.

The values are returned as a pandas `DataFrame` with these columns:
 - the value of the key
 - time of commit
 - commit_id
 - optionally, the label assigned to commit

The list is sorted by the time of commit in ascending order.

### CODE REVIEW

File: entropylab/api/param_store.py [hunk 1]
```diff
@@ -73,6 +71,21 @@ def list_commits(self, label):
         """
         pass
 
+    @abstractmethod
+    def list_values(self, key: str) -> List[Tuple[Any, datetime, str, str]]:
```
Comment (qguyk) [inline entropylab/api/param_store.py:75]: @urig I think that the return type here should be of a type that is easily plotted. maybe return a tuple of lists? or a numpy array?
worth adding a test that cheks you can get just the list of the values without the metadata...

@galwiner ?

**Review by qguyk — CHANGES_REQUESTED** [review]

File: entropylab/api/param_store.py (hunk 1 above)
Comment (galwiner) [reply entropylab/api/param_store.py:75]: An easy way to get what Guy is suggesting is the following: 
```python
results=pd.DataFrame([(1,2,'c',4),(2,3,'a','b'),(10,3,'a','b'),(20,3,'a','b')])
```
this then allows to slice through: 
```
results[0]
```
an even better experience is obtained if we give text names to the data frame columns

Comment (urig) [reply entropylab/api/param_store.py:75]: `list_values()` now returns a `pd.DataFrame` with named columns. 👍

Comment (qguyk) [inline entropylab/api/param_store.py:75]: @urig can you update the type hint here as well?

**Review by qguyk — CHANGES_REQUESTED** [review]

File: entropylab/api/param_store.py (hunk 1 above)
Comment (urig) [reply entropylab/api/param_store.py:75]: Tx!

Just a thought: is there a tool that can warn about mistakes like this?

**Review by galwiner — APPROVED** [review]

**Pull request merged**